src/runtime_src/core/common/smi.h

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+// Local include files
+#include "config.h"
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <string>
+#include <vector>
+#include <tuple>
+
+namespace xrt_core::smi {
+
+using TupleVector = std::vector<std::tuple<std::string, std::string, std::string>>; 
+
+struct basic_option {
+  std::string name;
+  std::string description;
+  std::string type;
+};
+
+struct option : public basic_option {
+  std::string alias;
+  std::string default_value;
+  std::string value_type;
+  std::vector<basic_option> description_array;
+
+  option(const std::string& name, 
+         const std::string& alias, 
+         const std::string& description,
+         const std::string& type, 
+         const std::string& default_value, 
+         const std::string& value_type, 
+         const std::vector<basic_option>& description_array = {})
+      : basic_option{name, description, type}, 
+        alias(alias), 
+        default_value(default_value), 
+        value_type(value_type), 
+        description_array(description_array) {}
+
+  boost::property_tree::ptree to_ptree() const;
+};
+
+/*
+* Each shim's smi class derives from this class
+* and adds its custom functionalities. Currently only validate tests and examine
+* reports differ between each shim but going forward, each shim can define its 
+* custom behavior for xrt-smi as required. This also gives us the flexibility
+* to add device specific xrt-smi behavior.
+*/
+
+class smi_base {
+protected:
+  virtual const TupleVector& get_validate_test_desc() const;
+  virtual const TupleVector& get_examine_report_desc() const;
+
+  std::vector<basic_option> construct_run_option_description() const;
+  std::vector<basic_option> construct_report_option_description() const;
+
+  boost::property_tree::ptree construct_validate_subcommand() const;
+  boost::property_tree::ptree construct_examine_subcommand() const;
+  boost::property_tree::ptree construct_configure_subcommand() const;
```

Comment:
I would prefer to see this as
```
virtual return_type
function() const;
```
I say this because code reading is easier when the important stuff is aligned to the left.

```
virtual const tuple_vector_type&
get_validate_test_desc() const;

std::vector<basic_option>
cosntruct_run_option_description() const;
```